WcaOnRails/app/webpacker/components/Delegates/index.jsx

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React from 'react';
+
+import { Button, Table } from 'semantic-ui-react';
+import I18n from '../../lib/i18n';
+import UserBadge from '../UserBadge';
+
+import '../../stylesheets/delegates/style.scss';
+
+const dasherize = (string) => string.replace(/_/g, '-');
+
+export default function Delegates({
+  delegates,
+  isEditVisible,
+}) {
+  const seniorDelegates = React.useMemo(() => delegates
+    .filter((user) => user.delegate_status === 'senior_delegate')
+    .sort((user1, user2) => user1.region.localeCompare(user2.region)), [delegates]);
+
+  // TO_VERIFY: I assume there are no cases where there are no delegates without
+  // senior delegate unless they are senior delegate
+
+  return seniorDelegates.map((seniorDelegate) => {
+    const delegatesUnderSenior = [seniorDelegate, ...delegates
+      .filter((user) => user.senior_delegate_id === seniorDelegate.id && user.delegate_status !== 'trainee_delegate')
```

Comment:
Why are we excluding Trainee Delegates?

One common nitpick that we sometimes get with the current Delegate page is that Trainees are not marked as such. We could use this rewrite as a motivation to include them, either by adding them in a separate section of the table below all Staff Delegates or by adding them inline with the other Delegates but using a different color.

Reply:
The main reason why trainee delegate is not added is because it is not there currently. 😅
And yes, adding it as part of this migration is possible, but I'm more excited to have the delegates table come into picture fast. If that happens, it will be more easier to show trainee delegates as well. I can add the trainee delegates in this PR, but I might have to change it completely once the next few PRs for this project comes.

Is it fine if I don't add it in this PR? I can promise that the next few PRs of delegate table project will come soon after merging this. In those PRs, trainee delegates will also come into the page.

Reply:
Agreed as per your suggestions above

Reply:
Thank you.